libs/ledger-live-common/src/deviceSDK/commands/firmwareUpdate/installFirmware.ts

```diff
@@ -0,0 +1,148 @@
+import { Observable, of, throwError } from "rxjs";
+import URL from "url";
+import Transport, { TransportStatusError } from "@ledgerhq/hw-transport";
+import type { FinalFirmware, OsuFirmware } from "@ledgerhq/types-live";
+import type { DeviceInfo, SocketEvent } from "@ledgerhq/types-live";
+import { version as livecommonversion } from "../../../../package.json";
+import { getEnv } from "../../../env";
+//import { createMockSocket, bulkSocketMock, secureChannelMock } from "../../api/socket.mock";
+import { log } from "@ledgerhq/logs";
+import { createDeviceSocket } from "../../../api/socket";
+import { catchError, filter, map } from "rxjs/operators";
+import {
+  ManagerFirmwareNotEnoughSpaceError,
+  UserRefusedFirmwareUpdate,
+  DeviceOnDashboardExpected,
+  ManagerDeviceLockedError,
+} from "@ledgerhq/errors";
+import { UnresponsiveCmdEvent } from "../core";
+
+export type InstallFirmwareCommandRequest = {
+  targetId: DeviceInfo["targetId"];
+  firmware: OsuFirmware | FinalFirmware;
+};
+
+type FilteredSocketEvent =
+  | {
+      type: "bulk-progress";
+      progress: number;
+      index: number;
+      total: number;
+    }
+  | {
+      type: "device-permission-requested";
+      wording: string;
+    };
+
+export type InstallFirmwareCommandEvent =
+  | {
+      type: "progress";
+      progress: number;
+    }
+  | {
+      type: "allowManagerRequested";
```

Comment:
side note: i think we wanted to change the "allow manager" vocabulary to "allow secure channel" - so `allowSecureChannelRequested`